src/errorpage.cc

```diff
@@ -694,8 +694,18 @@ ErrorState::ErrorState(err_type t, Http::StatusCode status, HttpRequest * req, c
     ale = anAle;
 }
 
-ErrorState::ErrorState(HttpRequest * req, HttpReply *errorReply) :
-    ErrorState(ERR_RELAY_REMOTE)
+ErrorState::ErrorState(err_type aType, Http::StatusCode status, char const *anUrl,
+                       Ip::Address &srcAddress, HttpRequest *aRequest,
+                       const AccessLogEntry::Pointer &al) : 
+                       ErrorState(aType, status, aRequest, al)
+{
+    src_addr = srcAddress;
```

Comment:
The ErrorState constructor should not demand two copies of the same information.

* If there is AccessLogEntry, it should contain the source address (whatever that is). If ALE does not contain that info today (and it is your responsibility, as a developer doing this refactoring, to check whether it does), the caller (or some caller-related code) should be fixed: ALE should get this info as soon as both ALE and the info become available.

* If there is no AccessLogEntry, then supplying `src_addr` explicitly might be OK, but, most likely, the caller should just work harder to get to the ALE (or even create it). ALE is becoming increasingly important to handling errors correctly. See, for example, 7e6eabb.

FWIW, there are other ongoing projects trying to polish/improve these ALE maintenance aspects in a more comprehensive manner.

Reply:
The objective of this iteration is to address your note and get rid of clientBuildError which, as you say, doesn't belong to client_side_request ; that's why I left the arguments in with the same order and names.

I'll leave a TODO behind noting to consolidate the constructors and minimise arguments if possible.

Reply:
similar argument stands for the aRequest parameter. Request should always be available via the ALE.

Reply:
> The objective of this iteration is to address your note

FWIW, the best way to address my concern, IMO, is to follow my suggestion: Keep clientBuildError(), but move it where it belongs. This approach keeps this Pandora Box closed and this PR focused on its primary goal.

I cannot insist on any specific path to victory, but I am pretty sure you do not fully realize the complexities involved in fixing this ALE-related code. Most likely, a proper solution will not be a quick follow up project, and such a project may conflict with other ongoing ALE-related efforts.

> I'll leave a TODO behind noting to consolidate the constructors and minimise arguments if possible, and fix it in a followup refactoring

FYI: Adding such a TODO to the constructor that should not have been added (at least not in this PR) would _not_ address my concern about adding that constructor. One cannot justify creating a brand new problem by adding a TODO to solve it when there is no need to add that problem in the first place.

Again, please consider undoing this out-of-scope change instead of trying to fix it or trying to convince me that you will fix it later.